File: Scala/src/main/scala/util/MultiplosDeDois.scala
package util

object MultiplosDeDois extends (Int => List[Int]) {

  override def apply(v1: Int): List[Int] = {
    if (v1 < 2) Nil
    else {
      if (v1%2==0) v1 :: (apply(v1-1))
      else apply(v1-1)
    }
  }

}

File: Java/src/util/MultiplosDeDois.java
package util;

import java.util.ArrayList;
import java.util.List;

public class MultiplosDeDois {

	private List<Integer> lista = new ArrayList<>();
	
	public List<Integer> multiplos(int v1) {
		for (; v1 < 2; v1--) {
			if (v1%2==0) {
				lista.add(v1);
			} else {
				continue;
			}
		}
		return lista;
	}
	
}
